### PR TITLE
Helm initialization of ClamAV virus scan

### PR DESCRIPTION
**Pull request by guru-aot**
- Copied Helm charts to our project from [source](https://github.com/bcgov/common-hosted-clamav-service)  
- Updated the values for our openshift namespace specific
- Created github actions to enable install, upgrade, and uninstall clamav using helm

Files that are changed
.github/workflows/clamav.yml
sims.code-workspace
devops/helm/main/values-0c27fb-dev.yaml
devops/helm/main/values-0c27fb-prod.yaml
devops/helm/main/values-0c27fb-test.yaml
devops/helm/main/values-a6ef19-dev.yaml
devops/helm/main/values-a6ef19-prod.yaml
devops/helm/main/values-a6ef19-test.yaml

Note: This is just a template for the github action, so once it is merged will create another PR if there are any changes needed in the github action.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,8 @@ testing/**/report/
 videos/
 cypress.env.json
 
+charts/
+template.yaml
+*.secret.yaml
+
+
```
Comment (andrewsignori-aot) [inline .gitignore:25]: Please emove the extra lines, or please clarify the reason to keep them.

File: .github/workflows/clamav.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    steps:
+      - name: Checkout Target Branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.gitRef }}
+      - name: Log in to OpenShift
+        run: |
+          oc login --token=${{ secrets.SA_TOKEN }} --server=${{ vars.OPENSHIFT_CLUSTER_URL }}
+      - name: Manage Clamav
+        working-directory: "./devops/helm/main/"
```
Comment (andrewsignori-aot) [inline .github/workflows/clamav.yml:44]: Is it not the case to create a dedicated folder from ClamAV under DevOps?
Or maybe we expected another similar "helm" something like `./devops/helm/clamAV`?

Comment (guru-aot) [reply .github/workflows/clamav.yml:44]: yeah we can close it in a folder and create one more for crunchy or any other

File: devops/helm/_clamav/values.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  cpu: 80
+  # average Memory usage utilization percentage per pod (1-100)
+  # memory: 80
+  # average http_requests utilization per pod (value as a string)
+  # requests: 1k
+
+podDisruptionBudget:
+  enabled: false
+  # minAvailable: 1
+  # maxUnavailable: 1
```
Comment (andrewsignori-aot) [inline devops/helm/_clamav/values.yaml:77]: Maybe not part of this PR, for clarification, `hpa` as `enabled: false` and `podDisruptionBudget` as `enabled: false` will require further investigations can we have them just enabled?

Comment (guru-aot) [reply devops/helm/_clamav/values.yaml]: yes we have to enable them, I will make changes in the upcoming PR

File: .github/workflows/clamav.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+name: Clamav - Install/Upgrade/Remove
+run-name: Clamav -  ${{ inputs.environment }} Clamav in ${{ inputs.environment }} using ${{ inputs.gitRef }}
```
Comment (andrewsignori-aot) [inline .github/workflows/clamav.yml:2]: Minor but I believe that we should call it ClamAV everywhere where it is not a variable name.

File: devops/helm/clamav/main/templates/networkpolicy.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-same-namespace
```
Comment (andrewsignori-aot) [inline devops/helm/clamav/main/templates/networkpolicy.yaml:4]: is it not going to affect our existing NetworkPolicies named `allow-same-namespace`?

Comment (guru-aot) [reply devops/helm/clamav/main/templates/networkpolicy.yaml]: These things can be removed safeside, i just copy pasted the whole helm/main folder. This is not applied during the helm install/upgrade/uninstall of the clamAV server. So this should not affect our network configuration.

File: .github/workflows/clamav.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+jobs:
+  manageClamav:
+    name: ${{ inputs.environment }} ClamAV in ${{ inputs.environment }}
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    env:
+      NAMESPACE: ${{ secrets.OPENSHIFT_ENV_NAMESPACE }}
+    steps:
+      - name: Checkout Target Branch
+        uses: actions/checkout@v4
```
Comment (andrewsignori-aot) [inline .github/workflows/clamav.yml:37]: Thanks for using the updated version.

**Review by andrewsignori-aot — APPROVED** [review]
Great work. Approving to allow the work to continue and the GitHub actions to be tested.
